### PR TITLE
feat: use signals to update user settings over sessions

### LINKED ISSUE #24981: 🚀 Feature: Add signals support for UserSettingsStorage
### 🔖 Feature description

As the signals are now stable in Backstage, it could be used to update the user settings in the frontend. 

### 🎤 Context

See https://github.com/backstage/backstage/blob/master/plugins/user-settings/src/apis/StorageApi/UserSettingsStorage.ts#L150

### ✌️ Possible Implementation

1. Add signals support for `user-settings-backend` to notify about user setting changes to the user
2. Add `useSignals` hook to the `UserSettingsStorage` to listen to changes in user settings from signals channel. The payload does not need to contain the new value but instead, just make the storage implementation to refetch the data from REST API.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

closes #24981

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/user-settings-backend/src/service/router.ts [hunk 1]
```diff
@@ -104,6 +108,14 @@ export async function createRouterInternal(options: {
       key,
     });
 
+    if (options.signals) {
+      await options.signals.publish<{ key: string }>({
+        recipients: { type: 'user', entityRef: userEntityRef },
+        channel: `user-settings`,
+        message: { key },
```
Comment (freben) [inline plugins/user-settings-backend/src/service/router.ts:115]: I don't know the common granularity people are meant to use here, but i'd expected some form of typing. Either

```
channel: 'user-settings:key-changed'
```

or

```
message: { type: 'setting-changed', key }
```

File: plugins/user-settings-backend/src/service/router.ts [hunk 2]
```diff
@@ -104,6 +108,14 @@ export async function createRouterInternal(options: {
       key,
     });
 
+    if (options.signals) {
+      await options.signals.publish<{ key: string }>({
```
Comment (freben) [inline plugins/user-settings-backend/src/service/router.ts:112]: signals generally would deserve to have dedicated named types right? like

```ts
export interface SignalKeyChanged {
  type: 'key-changed',
  key: string;
}
```

exported from a -common package, right?

Comment (drodil) [reply plugins/user-settings-backend/src/service/router.ts:112]: Yeah, that requires a completely new plugin as there's not `user-settings-common` available yet.

**Review by freben — APPROVED** [review]
Nice.

**Pull request merged**